.eslintrc.json

```diff
@@ -0,0 +1,42 @@
+{
+  "extends": "./node_modules/gts",
+  "env": {
+    "browser": true,
+    "node": true
+  },
+  "plugins": ["node", "ban"],
+  "rules": {
+    "ban/ban": [
+      "error",
+      {
+        "name": "setTimeout",
+        "message": "WPT (Web Platform Tests) doesn't allow setTimeout."
+      }
+    ],
+    "linebreak-style": ["error", "unix"],
+    "no-console": "error",
+    "@typescript-eslint/typedef": [
+      "error",
+      {
+        "arrowParameter": false,
+        "memberVariableDeclaration": false
+      }
+    ],
+    "@typescript-eslint/no-inferrable-types": "off",
+
+    "sort-imports": "warn",
+    "no-useless-escape": "warn",
+    "no-empty": "warn",
+    "@typescript-eslint/no-unused-vars": "warn",
+    "no-inner-declarations": "warn"
```

Comment:
@kainino0x what do you think? These several rules I have as "warn" right now because they're a bit different from what we had in TSLint.

Reply:
I'd like to keep master warning-free if we can, but I'm okay with doing lint autofixes across the codebase if they're not too huge

Reply:
sort-imports and no-useless-escape should definitely be okay to autofix

Reply:
I don't think any are actually autofixable, but here's what would change:

`no-useless-escape`: `\/` --> `/`

`sort-imports`: this one means sorting the import names, not the import filepaths which is what TSLint did. However, multi-imports take precedence. so:
```js
import { X, b, c } from './foo' // multi first
import { A } from './a' // then capital
import { d } from './d' // then lowercase
```
Most of the `@typescript-eslint/no-unused-vars` errors are from `g.test('foo', t => ...)` where `t` isn't used in the body. Most of these are unittests or examples, so I think we can /* eslint-disable-next-line */
 
`no-empty`: This is happening on our empty catch statements. We could use `allowEmptyCatch`, but there's only a couple right now that are very intentional. I think we should suppress this one at the site.

`no-inner-declarations`: functions need to be declared at the root scope.  The couple places we have to do this could be arrow functions instead which sounds good to me.

Reply:
Those all sound good to me. I'm surprised they don't have autofixers, especially sort-imports. It's going to be a pain if we have to manually fix our import sorting on every patch.